src/components/Button/Button.tsx

```diff
@@ -12,61 +12,59 @@ import {
 } from './Button.style';
 import Loader from 'components/Loader';
 
-export type Props = ButtonBaseProps;
+export type Props = ButtonBaseProps & TestProps & onClickProp;
 type onClickProp = { onClick: ClickHandler };
 
-const Button = React.forwardRef<HTMLButtonElement, ButtonProps & Props & TestProps & onClickProp>(
-  (props, ref) => {
-    const {
-      size = 'md',
-      type = 'primary',
-      color = '',
-      filled = true,
-      transparent = false,
-      iconLeft = null,
-      iconRight = null,
-      disabled = false,
-      children,
-      onClick,
-    } = props;
-    const { loading, handleAsyncOperation } = useLoading(onClick);
-    const childrenWrapperRef = useRef<HTMLSpanElement>(null);
-    const innerButtonWidth = childrenWrapperRef?.current?.clientWidth;
+const Button = React.forwardRef<HTMLButtonElement, Props & ButtonProps>((props, ref) => {
```

Comment:
why test props are missing?

Reply:
They were included in the Props type, defined on line 15 :)